src/Microsoft.AspNet.Diagnostics/StatusCodePagesMiddleware.cs

```diff
@@ -25,8 +25,17 @@ public StatusCodePagesMiddleware(RequestDelegate next, [NotNull]StatusCodePagesO
 
         public async Task Invoke(HttpContext context)
         {
+            context.SetFeature<IStatusCodeFeature>(new StatusCodeFeature());
+
             await _next(context);
 
+            var statusCodeFeature = context.GetFeature<IStatusCodeFeature>();
+            if (statusCodeFeature != null && !statusCodeFeature.Enabled)
```

Comment:
So you could disable the middleware by just removing the feature interface? That makes the Enabled property redundant. Hmm, that would reduce that section of the sample to `context.SetFeature<IStatusCodeFeature>(null);`

Is there any benefit to knowing that the application is capable of this feature, but that it is currently disabled?